R/fread.R

```diff
@@ -340,7 +340,7 @@ yaml=FALSE, autostart=NA, tmpdir=tempdir(), tz="UTC")
     if (!is.character(key))
       stopf("key argument of data.table() must be a character vector naming columns (NB: col.names are applied before this)")
     if (length(key) == 1L) {
-      key = strsplit(key, split = ",", fixed = TRUE)[[1L]]
+      if (key != strsplit(key,split=",")[[1L]]) stopf("Usage of string-separated literals in %s is deprecated, please split such entries yourself before passing to data.table", "key=")
```

Comment:
change string-separated to comma-separated?
do you want to add a test for this error message?